libs/core/src/components/pds-popover/pds-popover.tsx

```diff
@@ -0,0 +1,162 @@
+ * @slot content - HTML content for the popover
+ */
+
+@Component({
+  tag: 'pds-popover',
+  styleUrls: ['../../global/styles/base.scss', 'pds-popover.scss'],
+  shadow: true,
+})
+export class PdsPopover {
+  private contentEl: HTMLElement | null;
+  /**
+   * Reference to the Host element
+   */
+  @Element() el: HTMLPdsPopoverElement;
+
+  /**
+   * Determines when the popover is open
+   * @defaultValue false
+   */
+  @State() isOpen = false;
+
+  /**
+   * A unique identifier used for the underlying component `id` attribute.
+   */
+  @Prop() componentId: string;
+
+  /**
+   * Determines whether or not the popover has an arrow
+   * @defaultValue true
+   */
+  @Prop() hasArrow? = true;
+
+  /**
+   * Determines the preferred position of the popover
+   * @defaultValue "right"
+   */
+  @Prop({ reflect: true }) placement:
+    'top'
+    | 'top-start'
+    | 'top-end'
+    | 'right'
+    | 'right-start'
+    | 'right-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'bottom-end'
+    | 'left'
+    | 'left-start'
+    | 'left-end' = 'right';
+
+  /**
+   * Emitted after a popover is closed
+   */
+  @Event() pdsPopoverHide: EventEmitter;
+
+  /**
+   * Emitted after a popover is shown
+   */
+  @Event() pdsPopoverShow: EventEmitter;
+
+  componentDidRender() {
+    positionTooltip({elem: this.el, elemPlacement: this.placement, overlay: this.contentEl});
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.handleOutsideClick);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.handleOutsideClick);
+  }
+
+  @Listen('click', {
+    capture: true
+  })
+  handleClick(event: MouseEvent) {
+    const closestTarget = (event.target as HTMLElement).closest("div[slot='trigger']");
+
+    if (!closestTarget) {
+      event.stopPropagation();
+      return;
+    }
+
+    if (!this.isOpen) {
+      this.show();
+    } else {
+      this.hide();
+    }
+    event.stopPropagation();
+  }
+
+  @Listen('click', {
+    target: 'document'
+  })
+  handleOutsideClick(event: MouseEvent) {
+    if (this.isOpen) {
+      if(this.contentEl && !this.contentEl.contains(event.target as Node)) {
+        this.hide();
+      }
+    }
+  }
+
+  /**
+   * Shows the popover by enabling the isOpen state
+   */
+  @Method()
+  async show() {
+    this.isOpen = true;
+    this.pdsPopoverShow.emit();
+  }
+
+  /**
+   * Hides the popover by disabling the isOpen state
+   */
+  @Method()
+  async hide() {
+    this.isOpen = false;
+    this.pdsPopoverHide.emit();
+  }
+
+  render() {
+    return (
+      <Host
+      >
+        <div
+          class={`
+            pds-popover
+            pds-popover--${this.placement}
+            ${this.isOpen ? 'pds-popover--is-open' : ''}
+            ${this.hasArrow ? '' : 'pds-popover--no-arrow'}
+          `}
+        >
+          <span
+            aria-describedby={this.componentId}
+            class="pds-popover__trigger"
+            onClick={this.handleClick}
```

Comment:
There could be a better approach, but onClick excludes keyboard navigation because it's not a button or a link, right? Could we try something like this GPT recommended code? Specifically, also allow a keyboard to navigate to the popover? 

```suggestion
            onKeyDown={(event) => {
               if (event.key === "Enter" || event.key === " ") {
                     event.preventDefault(); // Prevents scrolling when pressing space
                     this.handleClick();
                   }
                 }}
                 role="button"
                 tabIndex="0" // Makes the span focusable for keyboard users
```
 
Totally open to other approaches!